src/header.sh

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+TEMPDIR="$(mktemp -d)"
+USAGE="
+Usage: $0 [OPTIONS]
+
+Arguments:
+  Path to an environment packed using pixi-pack
+
+Options:
+  -o, --output-directory <DIR>    Where to unpack the environment. The environment will be unpacked into a subdirectory of this path [default: env]
+  -e, --env-name <NAME>           Name of the environment [default: env]
+  -s, --shell <SHELL>             Sets the shell [options: bash, zsh, xonsh, cmd, powershell, fish, nushell]
+  -v, --verbose                   Increase logging verbosity
+  -q, --quiet                     Decrease logging verbosity
+  -h, --help                      Print help
+"
+
+# Check for help flag
+for arg in "$@"; do
+  if [ "$arg" = "-h" ] || [ "$arg" = "--help" ]; then
+    echo "$USAGE"
+    exit 0
+  fi
+done
+
+archive_begin=$(grep -anm 1 "^@@END_HEADER@@" "$0" | awk -F: '{print $1}')
+archive_end=$(grep -anm 1 "^@@END_ARCHIVE@@" "$0" | awk -F: '{print $1}')
+
+if [ -z "$archive_begin" ] || [ -z "$archive_end" ]; then
+  echo "ERROR: Markers @@END_HEADER@@ or @@END_ARCHIVE@@ not found." >&2
+  exit 1
+fi
+
+archive_begin=$((archive_begin + 2))
+archive_end=$((archive_end - 1))
+pixi_pack_start=$(($archive_end + 2))
+
+echo $(tail -n +$archive_begin "$0" | head -n $(($archive_end - $archive_begin + 1))) | base64 -d > "$TEMPDIR/archive.tar"
+echo $(tail -n +$pixi_pack_start "$0") | base64 -d > "$TEMPDIR/pixi-pack"
```

Comment:
this is not the most efficient way of doing this...

```
❯ time tail -n +64 environment.sh | head -n 1 | base64 -d > /tmp/archive.tar

________________________________________________________
Executed in   22.93 secs    fish           external
   usr time   20.63 secs    0.13 millis   20.63 secs
   sys time    1.29 secs   12.02 millis    1.28 secs
```

takes 22s on my macbook

while

```
❯ time sed -n "64p" environment.sh | base64 -d > /tmp/archive.tar

________________________________________________________
Executed in    7.45 secs    fish           external
   usr time    6.10 secs    0.11 millis    6.10 secs
   sys time    0.86 secs    5.41 millis    0.86 secs
```

takes only 7s.

does this sed command work with both macos and linux?

Reply:
I tried it on BSD and read through gnu's doc. It should be fine.

Reply:
this only works though, if the archive base64 is in a singe line (which it is, right?)
(we should assert this in the shell script somewhere

Reply:
Yes in our case it is. I'll do that. But also why only one line? I think this works for multiple lines as well, no?
```sh
sed -n "$i,$jp" 
```